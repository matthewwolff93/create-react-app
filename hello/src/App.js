import React from 'react';
import HelloWorldList from './HelloWorldList';

import './App.css';

const App = () => (
	<div className="App">
		<HelloWorldList />
	</div>
);

export default App;