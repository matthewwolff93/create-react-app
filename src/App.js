import React from 'react';
import HelloWorldList from './HelloWorldList';

import styles from './App.css';

const App = () => (
	<div className={styles.App}>
		<HelloWorldList />
	</div>
);

export default App;