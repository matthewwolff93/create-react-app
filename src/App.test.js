import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';
import HelloWorld from './HelloWorld';
import HelloWorldList from './HelloWorldList';
import AddGreeter from './AddGreeter';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<App />, div);
	ReactDOM.unmountComponentAtNode(div);
});

describe(HelloWorld, () => {
	const name = 'Person';
	const mockRemoveGreeting = jest.fn();
	const component = shallow(
		<HelloWorld name={name} removeGreeting={mockRemoveGreeting} />
	);

	it('renders and matches our snapshot', () => {
		const component = renderer.create(
			<HelloWorld name="Person" />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('contains the supplied name', () => {
		expect(component.text()).toContain(name);
	});

	it('modifies the greeting when frenchify is clicked', () => {
		component.find('button.frenchify').simulate('click');
		expect(component.text()).toContain('Bonjour');
	});
	
	it('calls the passed in removeGreeting function when remove button is clicked', () => {
		component.find('button.remove').simulate('click');
		expect(mockRemoveGreeting).toBeCalled();
	});
});

describe(HelloWorldList, () => {
	const component = shallow(
		<HelloWorldList />
	);

	it('renders and matches our snapshot', () => {
		const component = renderer.create(
			<HelloWorldList />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('contains an AddGreeter subcomponent', () => {
		expect(component.find(AddGreeter)).toHaveLength(1);
	});

	it('contains the same number of HelloWorld components as greetings', () => {
		const helloWorlds = component.find(HelloWorld).length;
		const greetings = component.state('greetings').length;
		expect(helloWorlds).toEqual(greetings);
	});

	it('adds another greeting when the add greeting function is called', () => {
		const before = component.find(HelloWorld).length;
		component.instance().addGreeting('Sample');
		component.update();
		const after = component.find(HelloWorld).length;
		expect(after).toBeGreaterThan(before);
	});

	it('removes a greeting from the list when the remove greeting function is called', () => {
		const before = component.find(HelloWorld).length;
		const removeMe = component.state('greetings')[0];
		component.instance().removeGreeting(removeMe);
		component.update();
		const after = component.find(HelloWorld).length;
		expect(after).toBeLessThan(before);
	});
});

describe(AddGreeter, () => {
	const mockAddGreeter = jest.fn();
	const component = shallow(
		<AddGreeter addGreeting={mockAddGreeter} />
	);

	it('renders and matches our snapshot', () => {
		const component = renderer.create(
			<AddGreeter />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('contains the form', () => {
		expect(component.find('input')).toHaveLength(1);
		expect(component.find('button')).toHaveLength(1);
	});

	it('calls the passed in addGreeting function when add button is clicked', () => {
		component.find('button').simulate('click');
		expect(mockAddGreeter).toBeCalled();
	});

	it('updates the form when keys are pressed', () => {
		const updateKey = 'foo';
		component.instance().handleUpdate({ target: { value: updateKey }});
		expect(component.state('greetingName')).toEqual(updateKey);
	});

	it('blanks out the greetingName when the button is clicked', () => {
		const updateKey = 'a';
		component.instance().handleUpdate({ target: { value: updateKey } });
		expect(component.state('greetingName')).toEqual(updateKey);
		component.find('button').simulate('click');
		expect(component.state('greetingName')).toHaveLength(0);
	});
});
