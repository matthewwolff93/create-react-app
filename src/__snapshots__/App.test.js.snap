// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class AddGreeter extends _react.Component {

	constructor(props) {
		super(props);
		this.state = { greetingName: '' };
		this.handleUpdate = this.handleUpdate.bind(this);
		this.addGreeting = this.addGreeting.bind(this);
	}

	handleUpdate(event) {
		this.setState({ greetingName: event.target.value });
	}

	addGreeting() {
		this.props.addGreeting(this.state.greetingName);
		this.setState({ greetingName: '' });
	}

	render() {
		return (
			_react2.default.createElement('div', { className: 'AddGreeter', __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this },
				_react2.default.createElement('input', { type: 'text', onChange: this.handleUpdate, value: this.state.greetingName, __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this }), '\\xA0\\xA0',

				_react2.default.createElement('button', { onClick: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this }, 'Add')));


	}} renders and matches our snapshot 1`] = `
<div
  className="AddGreeter"
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
    
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;

exports[`class HelloWorld extends _react.Component {

	constructor(props) {
		super(props);
		this.state = { greeting: 'Hello' };
		this.frenchify = this.frenchify.bind(this);
		this.removeGreeting = this.removeGreeting.bind(this);
	}

	frenchify() {
		this.setState({ greeting: 'Bonjour' });
	}

	removeGreeting() {
		this.props.removeGreeting(this.props.name);
	}

	render() {
		return (
			_react2.default.createElement('div', { className: 'HelloWorld', __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this },
				this.state.greeting, ' ', this.props.name, '!',
				_react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }),
				_react2.default.createElement('button', { className: 'frenchify', onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this }, 'Frenchify!'),
				_react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 28 }, __self: this }),
				_react2.default.createElement('button', { className: 'remove', onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 29 }, __self: this }, 'Remove Me!')));


	}} renders and matches our snapshot 1`] = `
<div
  className="HelloWorld"
>
  Hello
   
  Person
  !
  <br />
  <button
    className="frenchify"
    onClick={[Function]}
  >
    Frenchify!
  </button>
  <br />
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove Me!
  </button>
</div>
`;

exports[`class HelloWorldList extends _react.Component {

	constructor(props) {
		super(props);
		this.state = { greetings: ['Jim', 'Sally', 'Bender'] };
		this.addGreeting = this.addGreeting.bind(this);
		this.removeGreeting = this.removeGreeting.bind(this);
	}

	renderGreetings() {
		return this.state.greetings.map(name =>
		_react2.default.createElement(_HelloWorld2.default, { key: name, name: name, removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this }));

	}

	addGreeting(newName) {
		this.setState({ greetings: [...this.state.greetings, newName] });
	}

	removeGreeting(removeName) {
		const filteredGreetings = this.state.greetings.filter(name =>
		name !== removeName);

		this.setState({ greetings: filteredGreetings });
	}

	render() {
		return (
			_react2.default.createElement('div', { className: 'HelloWorldList', __source: { fileName: _jsxFileName, lineNumber: 35 }, __self: this },
				_react2.default.createElement(_AddGreeter2.default, { addGreeting: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 36 }, __self: this }),
				this.renderGreetings()));


	}} renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <div
    className="AddGreeter"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
      
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Jim
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Sally
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Bender
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
</div>
`;
